Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -4887,7 +4887,7 @@ ScrollingNodeID RenderLayerCompositor::registerScrollingNodeID(ScrollingCoordina
     if (nodeType == ScrollingNodeType::Subframe && !treeState.parentNodeID)
         nodeID = scrollingCoordinator.createNode(nodeType, nodeID);
     else {
-        auto newNodeID = scrollingCoordinator.insertNode(nodeType, nodeID, treeState.parentNodeID.value_or(0), treeState.nextChildIndex);
+        auto newNodeID = scrollingCoordinator.insertNode(nodeType, nodeID, treeState.parentNodeID.value_or(ObjectIdentifier<ScrollingNodeIDType>(0)), treeState.nextChildIndex);
```

Comment:
`insertNode()` should use `std::optional<ScrollingNodeID>` for the parentNodeID.

Reply:
Fixed.

Reply:
Hmm actually this is turning into quite a big patch, since if we introduce std::optional here we have to do it everywhere, since if we set the optional to a 0 scrolling node id converting it to bool would return true rather than false. I think this should be addressed in a follow up patch as well.